examples/jsondiff/check_test.go

```diff
@@ -0,0 +1,137 @@
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+}
+
+func TestSubsetOfJSON(t *testing.T) {
+	// This test validates that the SubsetOfJSON check functions work; it is not
+	// a direct example of usage.
+
+	const (
+		a = `{"foo": "A", "bar": "B"}`
+		b = `{"foo": "A", "bar": "C"}`
+		c = `{"foo": "A"}`
+	)
+
+	t.Run("when a is cheked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(a)
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when []byte(a) is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)([]byte(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when json.RawMessage(a) is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(json.RawMessage(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when b is checked against a", func(t *testing.T) {
+		err := pkg.SubsetOfJSON(a)(b)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+
+	t.Run("when c is checked against a", func(t *testing.T) {
+		// check if c is subset of a; normal syntax would be:
+		// t.Run(name, subtest.Value(c).Test(pkg.SubsetOfJSON(a))
+		err := pkg.SubsetOfJSON(a)(c)
+		t.Run("then the check should fail", subtest.Value(err).NoError())
+	})
+	t.Run("when a is cheked against c", func(t *testing.T) {
+		// check if a is subset of c; normal syntax would be:
+		// t.Run(name, subtest.Value(a).Test(pkg.SubsetOfJSON(c))
+		err := pkg.SubsetOfJSON(c)(a)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+}
+
+func TestSupersetOfJSON(t *testing.T) {
+	// This test validates that the SupersetOfJSON check functions work; it is
+	// not a direct example of usage.
+
+	const (
+		a = `{"foo": "A", "bar": "B"}`
+		b = `{"foo": "A", "bar": "C"}`
+		c = `{"foo": "A"}`
+	)
+
+	t.Run("when a is cheked against a", func(t *testing.T) {
+		err := pkg.SupersetOfJSON(a)(a)
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when []byte(a) is checked against a", func(t *testing.T) {
+		err := pkg.SupersetOfJSON(a)([]byte(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when json.RawMessage(a) is checked against a", func(t *testing.T) {
+		err := pkg.SupersetOfJSON(a)(json.RawMessage(a))
+		t.Run("then the check should pass", subtest.Value(err).NoError())
+	})
+
+	t.Run("when b is checked against a", func(t *testing.T) {
+		err := pkg.SupersetOfJSON(a)(b)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+
+	t.Run("when c is checked against a", func(t *testing.T) {
+		// check if c is superset of a; normal syntax would be:
+		// t.Run(name, subtest.Value(c).Test(pkg.SupersetOfJSON(a)))
+		err := pkg.SupersetOfJSON(a)(c)
+		t.Run("then the check should fail", subtest.Value(err).Error())
+	})
+	t.Run("when a is cheked against c", func(t *testing.T) {
```

Comment:
A is the SuperSet of C

Reply:
Improved the naming to be more clear. 

To be clear, we are using the following wording:
1. Given - the state of things before an action.
2. When - which action we are performing.
3. Then - what are the result(s) of the action.

Normally subtest functions wold be used for step 3, but when we are validating that they work, we instead use them  as part of step 2.

In other words, given `c` is a subset of `a`, we are testing _if_ `a` is decodable to JSON _and_ it is a JSON superset of `c`. After we have carried this out, the final `then` condition to compare the results.